From 608d39faed63ea212b6c6cdf9fe2bef92e2120ea Mon Sep 17 00:00:00 2001
From: whyisjake <whyisjake@git.wordpress.org>
Date: Mon, 14 Oct 2019 15:27:04 +0000
Subject: [PATCH] HTTP API: Protect against hex interpretation.

Return earlier from wp_http_validate_url().

Props: iandunn, xknown, voldemortensen, whyisjake.

Built from https://develop.svn.wordpress.org/trunk@46475


git-svn-id: http://core.svn.wordpress.org/trunk@46273 1a063a9b-81f0-0310-95a4-ce76da25c4cd
---
 wp-includes/http.php    | 2 +-
 wp-includes/version.php | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/wp-includes/http.php b/wp-includes/http.php
index 36922da60e26..3356b9716295 100644
--- a/wp-includes/http.php
+++ b/wp-includes/http.php
@@ -550,7 +550,7 @@ function wp_http_validate_url( $url ) {
 		} else {
 			$ip = gethostbyname( $host );
 			if ( $ip === $host ) { // Error condition for gethostbyname()
-				$ip = false;
+				return false;
 			}
 		}
 		if ( $ip ) {
diff --git a/wp-includes/version.php b/wp-includes/version.php
index a4f93bdb27f7..8805c87052db 100644
--- a/wp-includes/version.php
+++ b/wp-includes/version.php
@@ -13,7 +13,7 @@
  *
  * @global string $wp_version
  */
-$wp_version = '5.3-beta3-46474';
+$wp_version = '5.3-beta3-46475';
 
 /**
  * Holds the WordPress DB revision, increments when changes are made to the WordPress DB schema.
